waku/waku_rln_relay/group_manager/on_chain/group_manager.nim

```diff
@@ -528,3 +535,27 @@ method stop*(g: OnchainGroupManager): Future[void] {.async.} =
     error "failed to flush to the tree db"
 
   g.initialized = false
+
+proc isSyncing*(g: OnchainGroupManager): Future[bool] {.async,gcsafe.} =
+  let ethRpc = g.ethRpc.get()
+
+  try:
+    let syncing = await ethRpc.provider.eth_syncing()
+    return syncing.getBool()
+  except CatchableError:
+    error "failed to get the syncing status", error = getCurrentExceptionMsg()
+    return false
+
+method isReady*(g: OnchainGroupManager): Future[bool] {.async,gcsafe.} =
```

Comment:
perhaps its cleaner to run the checks for `true` all together?

```
if g.ethRpc.isSome() && g.lastSeenBlockHead !=0 && g.latestProcessedBlock >= g.lastSeenBlockHead && !await g.isSyncing():
  return true
return false
```

maybe just a personal prefernce, just a sugerence.

Reply:
Yeah, I did think of this, imo the way it is right now is slightly more readable